cli/lsp/cache.rs

```diff
@@ -49,13 +49,13 @@ impl CacheServer {
     let _join_handle = thread::spawn(move || {
       let runtime = create_basic_runtime();
       runtime.block_on(async {
-        let ps = ProcState::build(Arc::new(Flags {
```

Comment:
It's handled internally in ProcState. So it will put the `Flags` into an `Arc<RootConfig>`. Ideally, you put the flags into proc state and then deal with the `Arc<RootConfig>` from there.